### PR TITLE
Add README for kpod

### PR DESCRIPTION
**Pull request by mrunalp**
We have gotten feedback from discussions with kubernetes developers that we need a utility to debug pods and images when the daemons are wedged. This PR proposes kpod to fulfill the role.

Signed-off-by: Mrunal Patel <mpatel@redhat.com>

### CODE REVIEW

**Comment by Crazykev** [comment]
Is there any details about how to do that?

**Comment by mrunalp** [comment]
@Crazykev This is just a README to propose and kick this off. We may have to make some changes in state management to share state between ocid daemon and kpod. It will be some work but beneficial in the long run as with kpod we will get a daemonless way to debug pods. Today k8s users have to resort to learning about the container runtime that is in use. kpod aims to make the debugging process runtime agnostic atleast when using cri-o. If we implement useful functionality here we could propose this as standard debug UI to k8s.

**Comment by cyphar** [comment]
But since ocid is designed to be "daemonless" in the traditional meaning (you don't need to run it all the time) surely `kpod` would just be a different running mode for `ocid` -- like combining `ocid` and `ocic`?

**Comment by runcom** [comment]
> But since ocid is designed to be "daemonless" in the traditional meaning (you don't need to run it all the time) surely kpod would just be a different running mode for ocid -- like combining ocid and ocic?

I remember we talked about this once on IRC :smile: BTW, ocid *is* a daemon in this scenario as it must stay up and running to serve requests for the CRI.

what we aim to do with `kpod` is to have something which could simply run a pod on a host w/o requiring any interaction with `ocid` (or the `kubelet` as I understand it). Imagine `kpod` as something which talks to runc, conmon and the like to spawn pods, pull images and whatnot w/o ever asking ocid/kubelet something. It'll just run.

It *could* be a different running mode for `ocid` yes, hence having another tool/cmd `kpod` to achieve this. It'll share some code API we have in place with ocid/ocic.

**Comment by feiskyer** [comment]
> kpod aims to make the debugging process runtime agnostic atleast when using cri-o. If we implement useful functionality here we could propose this as standard debug UI to k8s.

@runcom So above comments is not true? k8s knows nothing about runc.

**Comment by runcom** [comment]
> @runcom So above comments is not true? k8s knows nothing about runc.

I may have expressed myself wrongly. Sorry, what I meant is that `kpod` will run a pod using what the CRI would have sent to `ocid` via the API. Instead `kpod` will run `sandboxrun` directly on a host. @mrunalp correct me if I'm wrong.

**Comment by feiskyer** [comment]
@runcom OK. It's better adding some design docs (explaining how it works) in the README.

**Comment by rhatdan** [comment]
I  believe the goal of kpod is a lot more then just running pods.  
We also need a manager to be able to do things like the following:

```
* Manage pods 
-  List/getattr running, all
-  Create/Start/Stop/exec into/remove
* Manage images
- List/Pull/remove/getattr Images
```

Lots of potential use cases.

**Comment by mrunalp** [comment]
@cyphar Yeah, it will be something like a combination of the ocid/ocic that can just run any pod and also address other management use cases as @rhatdan pointed out. 
@feiskyer I will update the README with some more design notes.

**Comment by feiskyer** [comment]
LGTM

File: cmd/kpod/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# kpod - Simple debugging tool for pods and images
+kpod is a simple client only tool to help with debugging issues when daemons such as CRI runtime and the kubelet are not responding or
+failing. A shared API layer could be created to share code between the daemon and kpod. kpod does not require any daemon running. kpod
+utilizes the same underlying components that ocid uses viz., containers/image, container/storage, oci-runtime-tool/generate, runc or
```
Comment (runcom) [inline cmd/kpod/README.md]: `viz.`?

Comment (rhatdan) [inline cmd/kpod/README.md]: I have never seen it before either,   But Wikipedia says it makes sense.

https://en.wikipedia.org/wiki/Viz.

Comment (rhatdan) [inline cmd/kpod/README.md]: I would probably use i.e.

Comment (rhatdan) [inline cmd/kpod/README.md]: Or spell it out to be easier for non native speakers, or dummies like me.  :^)

**Comment by mrunalp** [comment]
@rhatdan @runcom Updated to remove viz. :)

**Comment by runcom** [comment]
LGTM

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**